src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala

```diff
@@ -58,10 +60,22 @@ class QueryHandler(dataSource: PermissionsDataSource, implicit val ec: Execution
             entity: AgoraEntity,
             entityTypes: Seq[AgoraEntityType.EntityType],
             username: String,
-            onlyPayload: Boolean): Future[PerRequestMessage] = {
+            onlyPayload: Boolean,
+            payloadAsObject: Boolean): Future[PerRequestMessage] = {
     agoraBusiness.findSingle(entity, entityTypes, username: String) map { foundEntity =>
-      if (onlyPayload) RequestComplete(foundEntity.payload)
-      else RequestComplete(foundEntity)
+      if (onlyPayload && payloadAsObject) {
+        RequestComplete(BadRequest, "onlyPayload, payloadAsObject cannot be used together")
+      } else {
+
+        if (onlyPayload) {
+          RequestComplete(foundEntity.payload)
+        } else if (payloadAsObject) {
+          RequestComplete(foundEntity.deserializeConfigurationPayload)
```

Comment:
rename `deserializeConfigurationPayload` to `withDeserializedPayload` or similar? Seeing it here parallel to `foundEntity.payload` is confusing.